"make docker" command fails

**Bug description**
When running "GOOS=linux GOARCH=amd64  make docker", there is an error: 
make: *** No rule to make target /Users/leitang/go/src/istio.io/istio/out/linux_amd64/release/metadata-exchange-filter.wasm', needed by docker.proxyv2'.  Stop.

Based on the PR https://github.com/istio/istio/pull/20259, metadata-exchange-filter.wasm should be generated when the make command is executed. Since the make fails, this generation mechanism does not work.


[ ] Configuration Infrastructure
[ ] Docs
[X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ X] Developer Infrastructure

**Expected behavior**
"make docker" command succeeds

**Steps to reproduce the bug**
1. OS: MacOS. 
2. run "GOOS=linux GOARCH=amd64  make docker".

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**
MacOS.

--- a/Makefile.core.mk
+++ b/Makefile.core.mk
@@ -226,6 +226,7 @@ ${ISTIO_ENVOY_MACOS_RELEASE_PATH}: init
 depend: init | $(ISTIO_OUT)
 
 DIRS_TO_CLEAN := $(ISTIO_OUT)
+DIRS_TO_CLEAN += $(ISTIO_OUT_LINUX)
 
 $(OUTPUT_DIRS):
 	@mkdir -p $@

--- a/bin/init.sh
+++ b/bin/init.sh
@@ -225,9 +225,6 @@ fi
 
 # Donwload WebAssembly plugin files
 WASM_RELEASE_DIR=${ISTIO_ENVOY_LINUX_RELEASE_DIR}
-if [[ "$LOCAL_OS" == "Darwin" ]]; then
-  WASM_RELEASE_DIR=${ISTIO_ENVOY_MACOS_RELEASE_DIR}
-fi
 for plugin in stats metadata_exchange
 do
   FILTER_WASM_URL="${ISTIO_ENVOY_BASE_URL}/${plugin}-${ISTIO_ENVOY_VERSION}.wasm"

--- a/tools/istio-docker.mk
+++ b/tools/istio-docker.mk
@@ -93,6 +93,10 @@ else
 	cp ${ISTIO_ENVOY_LINUX_RELEASE_PATH} ${ISTIO_ENVOY_LINUX_RELEASE_DIR}/envoy
 endif
 
+# rule for wasm extensions.
+$(ISTIO_ENVOY_LINUX_RELEASE_DIR)/stats-filter.wasm: init
+$(ISTIO_ENVOY_LINUX_RELEASE_DIR)/metadata-exchange-filter.wasm: init
+
 # Default proxy image.
 docker.proxyv2: BUILD_PRE=&& chmod 755 envoy pilot-agent istio-iptables
 docker.proxyv2: BUILD_ARGS=--build-arg proxy_version=istio-proxy:${PROXY_REPO_SHA} --build-arg istio_version=${VERSION} --build-arg BASE_VERSION=${BASE_VERSION}

